var/spack/repos/builtin/packages/py-tomopy/package.py

```diff
@@ -13,9 +13,24 @@ class PyTomopy(PythonPackage):
 
     homepage = "https://tomopy.readthedocs.io/en/latest/index.html"
     url      = "https://github.com/tomopy/tomopy/archive/1.0.0.tar.gz"
+    git      = 'https://github.com/tomopy/tomopy.git'
 
+    version('master', branch='master')
+    version('1.10.4', sha256='2f15edda11b4337a1a5560684fcf8f28a35c5035932b22b842ce728bd13fba01')
     version('1.0.0', sha256='ee45f7a062e5a66d6f18a904d2e204e48d85a1ce1464156f9e2f6353057dfe4c')
 
+    # GPU accel needs PTL which is a git submodule. Thus, we can only build it on master
+    depends_on('cuda',                          when='@master')
+    # The shared opencv is not found by during runtest. Not using GOT/PLT is faster too
+    depends_on('opencv+imgproc~shared',         when='@master')
+    # During the runtest, the shared MKL libs aren't found yet:
+    # depends_on('intel-mkl~shared')
+    depends_on('cmake@3.17:',                   type=('build'))
+    depends_on('ninja',                         type=('build'))
+    depends_on('py-setuptools-scm-git-archive', type=('build'))
```

Comment:
Missing dep on `py-setuptools-scm`

Reply:
py-setuptools-scm-git-archive is enough, it builds with the these depends_on just find and I don't have any setuptools installed, neither as Ubuntu dpkg, nor thru pip3, nor manually.

Reply:
`py-setuptools-scm-git-archive` has a transitive dep on `py-setuptools-scm`, which is why this currently builds. However, we may remove transitive deps from the build environment in the future, so you should always explicitly list direct dependencies even if they are already transitive dependencies.

Reply:
Ok, added.